test/e2e/network_segmentation.go

```diff
@@ -612,8 +612,11 @@ var _ = Describe("Network Segmentation", func() {
 					By("delete pods in test namespace to unblock CUDN CR & associate NAD deletion")
 					_, err := e2ekubectl.RunKubectl(c.namespace, "delete", "pod", "--all")
 					Expect(err).NotTo(HaveOccurred())
-					_, err = e2ekubectl.RunKubectl("", "delete", "clusteruserdefinednetwork", c.name)
-					Expect(err).NotTo(HaveOccurred())
+					_, _ = e2ekubectl.RunKubectl("", "delete", "clusteruserdefinednetwork", c.name)
+					Eventually(func() bool {
+						_, err := e2ekubectl.RunKubectl("", "get", "clusteruserdefinednetwork", c.name)
+						return strings.Contains(err.Error(), fmt.Sprintf("clusteruserdefinednetworks.k8s.ovn.org %q not found", c.name))
+					}, 1*time.Minute, 3*time.Second).Should(BeTrueBecause("ClusterUserDefinedNetwork %q should be gone", c.name))
```

Comment:
should we use the `MatchError` matcher instead ? It works with [string matchers internally](https://github.com/onsi/gomega/blob/f1ff459a1aa4fb0000c65d4051b09d669a09d346/matchers.go#L127).

Meaning, I **think** the following would work:

```suggestion
					Eventually(func() error {
						_, err := e2ekubectl.RunKubectl("", "get", "clusteruserdefinednetwork", c.name)
						return err 
					}, 1*time.Minute, 3*time.Second).Should(MatchError(ContainSubstring(fmt.Sprintf("clusteruserdefinednetworks.k8s.ovn.org %q not found", c.name))))
```

Reply:
Done